react/features/prejoin/components/web/Prejoin.tsx

```diff
@@ -212,7 +212,8 @@ const Prejoin = ({
     updateSettings: dispatchUpdateSettings,
     videoTrack
 }: IProps) => {
-    const showDisplayNameField = useRef(canEditDisplayName || showErrorOnJoin);
+    const showDisplayNameField = useMemo(
+        () => isDisplayNameVisible && showErrorOnJoin, [ isDisplayNameVisible, showErrorOnJoin ]);
```

Comment:
What do we show in the error? It looks like now it will not be shown if the display name is not visible. Is that ok?

Reply:
> What do we show in the error?

"Please enter your name to join the meeting"

You still get the prompt. I need to check again, but there was a weird situation created by the OR operator here. I'm not entirely sure what purpose the readonly display name serves. Because you might want to change your display name without having to clear.. local storage.

Reply:
IIRC we make it read only if it comes from a token.

Reply:
I don't think that's how it actually works atm, but we can fix that.